x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -451,35 +456,214 @@ export interface FetchAlertData {
   ecsAlertsData: unknown[];
 }
 
-export type AlertsTableProps = {
-  alertsTableConfiguration: AlertsTableConfigurationRegistry;
-  cases: {
-    data: Map<string, Case>;
-    isLoading: boolean;
-  };
-  maintenanceWindows: {
-    data: Map<string, MaintenanceWindow>;
-    isLoading: boolean;
+type MergeProps<T, AP> = T extends (args: infer Props) => unknown
+  ? (args: Props & AP) => ReactNode
+  : T extends ComponentClass<infer Props>
+  ? ComponentClass<Props & AP>
+  : never;
+
+export interface SelectedAlertWithLegacyFormats {
+  alert: Alert;
+  /**
+   * @deprecated
+   */
+  legacyAlert: LegacyField[];
+  /**
+   * @deprecated
+   */
+  ecsAlert: any;
+}
+
+export interface AlertsTableProps<AC extends AdditionalContext = AdditionalContext>
+  extends PublicAlertsDataGridProps {
+  id: string;
+  featureIds: ValidFeatureId[];
+  columns?: EuiDataGridProps['columns'];
+  query: Pick<QueryDslQueryContainer, 'bool' | 'ids'>;
+  initialSort?: SortCombinations[];
+  initialPageSize?: number;
+  browserFields?: BrowserFields;
+  onUpdate?: (context: RenderContext<AC>) => void;
+  onLoaded?: (alerts: Alerts, columns: EuiDataGridColumn[]) => void;
+  runtimeMappings?: MappingRuntimeFields;
+  showAlertStatusWithFlapping?: boolean;
+  toolbarVisibility?: EuiDataGridToolBarVisibilityOptions;
+  /**
+   * Allows to consumers of the table to decide to highlight a row based on the current alert.
+   */
+  shouldHighlightRow?: (alert: Alert) => boolean;
+  /**
+   * Enable when rows may have variable heights (disables virtualization)
+   */
+  dynamicRowHeight?: boolean;
+  emptyStateHeight?: 'tall' | 'short';
+  /**
+   * An object that will be passed along with the renderContext to all render functions.
+   */
+  additionalContext?: AC;
+  /**
+   * Cell content render function
+   */
+  renderCellValue?: MergeProps<
+    EuiDataGridProps['renderCellValue'],
+    RenderContext<AC> & SelectedAlertWithLegacyFormats
+  >;
+  renderCellPopover?: MergeProps<
+    EuiDataGridProps['renderCellPopover'],
+    RenderContext<AC> & { alert: Alert }
+  >;
+  renderActionsCell?: MergeProps<
+    EuiDataGridControlColumn['rowCellRender'],
+    RenderContext<AC> &
+      SelectedAlertWithLegacyFormats & { setIsActionLoading?: (isLoading: boolean) => void }
+  >;
+  renderAdditionalToolbarControls?: ComponentRenderer<AC>;
+  renderFlyoutHeader?: FlyoutSectionRenderer<AC>;
+  renderFlyoutBody?: FlyoutSectionRenderer<AC>;
+  renderFlyoutFooter?: FlyoutSectionRenderer<AC>;
+
+  lastReloadRequestTime?: number;
+}
+
+/**
+ * A utility type to extract the type of a prop from `AlertsTableProps`, excluding `undefined`.
+ */
+export type AlertsTableProp<Key extends keyof AlertsTableProps> = NonNullable<
```

Comment:
nit: I found it a bit confusing that the one is without an `s` (`AlertsTableProp`) and the other with an `s` (`AlertsTableProps`). If one of them is internal we can prefix it with `_` or find another name. Wdyt?

Reply:
Oh, I see what you mean. I thought it was giving you all the alerts table props. Yes, `GetAlertsTableProp` sounds better to avoid confusion. But it is a nit. Up to you.